features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/timeline/TimelineEvents.kt

```diff
@@ -22,16 +22,23 @@ sealed interface TimelineEvents {
     data object LoadMore : TimelineEvents
     data class SetHighlightedEvent(val eventId: EventId?) : TimelineEvents
     data class OnScrollFinished(val firstIndex: Int) : TimelineEvents
+
+    // Events coming from a timeline item
+    sealed interface EventFromTimelineItem : TimelineEvents
+
+    // Events coming from a poll item
+    sealed interface TimelineItemPollEvents : EventFromTimelineItem
```

Comment:
Nit: maybe this should wrap all the existing `TimelineItemPollEvents` below?

Reply:
Yes, it could, and I did it first, but the problem is that is make the access more complex, since all the hierarchy has to be imported, or explicitly written. I wanted to keep it flat, even if at first sight, it's a bit harder to read the parents chain.

Reply:
I also thought about splitting `TimelinePresenter.handleEvents()` into smaller function, by casing the interfaces type, but we may consider this when we will have more Events, because for now it's just adding cognitive complexity.